src/view/screens/Home.tsx

```diff
@@ -32,9 +39,39 @@ import {HomeHeader} from '../com/home/HomeHeader'
 
 type Props = NativeStackScreenProps<HomeTabNavigatorParams, 'Home'>
 export function HomeScreen(props: Props) {
+  const {navigation} = props
+  const {hasSession} = useSession()
   const {data: preferences} = usePreferencesQuery()
   const {data: pinnedFeedInfos, isLoading: isPinnedFeedsLoading} =
     usePinnedFeedsInfos()
+  const activeStarterPack = useActiveStarterPack()
+  const setActiveStarterPack = useSetActiveStarterPack()
+  const {setShowLoggedOut, requestSwitchToAccount} = useLoggedOutViewControls()
+
+  React.useEffect(() => {
+    // This will be true if the app was launched with a starter pack referral.
+    if (activeStarterPack?.uri) {
+      if (hasSession) {
+        const parsed = parseStarterPackUri(activeStarterPack.uri)
+        if (!parsed) return
+        setActiveStarterPack(undefined)
+        navigation.navigate('StarterPack', parsed)
+      } else {
+        setShowLoggedOut(true)
+        requestSwitchToAccount({
+          requestedAccount: isWeb ? 'starterpack' : 'new',
```

Comment:
Nit: `new` is explicitly defined in `shell/logged-out` as a special case, would be nice to have `starterpack` in there too